### DISCUSSION TITLE
[Question]: How to secure keys and secrets used in Librechat Application

### DISCUSSION THREAD

**Discussion by geodanchev** [danny-avila/LibreChat#3446] [Q&A]
### What is your question?

Since it's not a best practice to use env vars for keys and secrets ( see this [article](https://blog.diogomonica.com//2017/03/27/why-you-shouldnt-use-env-variables-for-secret-data/)) is there a way to pass those settings securely ?

### More Details

My organization insists of having key vaults and environment variables are easily printed out with 'exec env' command.

### What is the main subject of your question?

Other

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
it totally depends on your environment. 

for example, to name a few:
```
    AWS Secrets Manager: A service that helps securely store and manage access to secrets needed by your applications.
    HashiCorp Vault: A robust tool for securely accessing secrets, providing dynamic secrets, and managing sensitive data.
    Azure Key Vault: Primarily for Microsoft Azure users, securely stores and manages secrets, keys, and certificates.
    Google Cloud Secret Manager: For users on Google Cloud, it stores API keys, passwords, and other sensitive data.
```

for local, as long as your local env is secure it's not really a concern. It's common practice to use .env for this

**Reply by geodanchev** [reply]
Thanks for the feedback. 
The question is - how to set up Librechat to read those key vaults values?
In the current situation I'm using Azure Key Vault, and I have written simple python app proving that I can read the values from the virtual machine that Librechat Ubuntu Docker Deployment is working. So now I'm ready just to pass that values securely to the docker containers and remove them from the .env files.

**Comment by InAnYan** [comment]
I'm also interested in this feature.

For example, many *desktop* applications use FreeDesktop Secrets (Keyring). Maybe because LibreChat is not a desktop application, secrets management could be troubling

**Reply by geodanchev** [reply]
My approach is to start the docker containers with these secrets in the docker compose command. This way they are not present into the .env file. If you want I can give you an example.
